src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -11,6 +11,19 @@ namespace System.Net
 {
     internal class IPAddressParser
     {
+
+        internal static bool ContainsColon(ReadOnlySpan<char> ipSpan)
+        {
+            for (var i = 0; i < ipSpan.Length; i++)
+            {
+                if (ipSpan[i] == ':')
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
Rather than writing this method, you should be able to add a reference to System.Memory in the .csproj:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Primitives/src/System.Net.Primitives.csproj#L200
Then you can delete this method, and at the call site instead of doing:
```C#
if (ContainsColon(ipSpan))
```
do:
```C#
if (ipSpan.IndexOf(':') >= 0)
```